### ISSUE TITLE
URL decoding values in authData

### ISSUE THREAD

**Issue by whatthewhat** [Vestorly/torii#323]
Hey 👋

I'm currently working on a salesforce torii provider (current progress is here: https://github.com/whatthewhat/torii-salesforce-oauth2) based on the `oauth2-code` base provider.
Salesforce returns a URL encoded authorization code that needs to be decoded before using it to obtain an access token.. *I'm not willing to admit how much time it took me to figure out that this was the reason the authentication was failing* 😆

Basically I'm wondering if you think it makes sense for torii to call `decodeURIComponent` on all values in the authentication hash by default (would be happy to work on a PR for that) or if it's more of a per-provider concern and I should override `open` in the SF provider to achieve this?

**Comment by bantic** [comment]
@whatthewhat Thanks for opening the issue! Sorry you got trolled by the encoded authorization code. I'm surprised salesforce would issue something other than an alphanumeric code. Yes, it probably is Torii's responsibility to decode the query params before presenting them to the user. A PR would be happily accepted. In the meantime, overriding your `open` as you describe would be the best method.

**Comment by whatthewhat** [comment]
@bantic Just opened a PR - https://github.com/Vestorly/torii/pull/376, hopefully, this will save someone in the future from getting trolled the same way :)
